backend-rust/src/migrations/m0005.sql

```diff
@@ -0,0 +1 @@
+CREATE INDEX transactions_hash_idx ON transactions USING gin (hash text_pattern_ops);
```

Comment:
Please add a comment why we introduce this index:

-- Used to efficiently perform partial string matching on the `hash` column,
-- allowing fast lookups when searching for transactions by their hash prefix using `LIKE`.